blocks/api/post.pegjs

```diff
@@ -148,7 +148,20 @@ WP_Block_End
   }
 
 WP_Block_Name
-  = $(ASCII_Letter (ASCII_AlphaNumeric / "/" ASCII_AlphaNumeric)*)
+  = blockName:$(ASCII_Letter (ASCII_AlphaNumeric / "/" ASCII_AlphaNumeric)*)
+  {
+    /** <?php
+    if ( false === strpos( $blockName, '/' ) ) {
+      $blockName = "core/$blockName";
+    }
+    return $blockName;
+    ?> **/
+
+    if ( ! blockName.includes( '/' ) ) {
+      blockName = 'core/' + blockName;
+    }
+    return blockName;
+  }
```

Comment:
if we're going to encode this here we might want to consider making these separate tokens in the grammar instead of adding more actual custom code in the parser

```
WP_Block_Name
  = WP_Core_Block_Name
  / WP_Extra_Block_Name

WP_Core_Block_Name
  = name:$(ASCII_Letter ASCII_AlphaNumeric*)
  { /** <?php return "core/$name" ?> */ return 'core/' + name }

WP_Extra_Block_Name
  = $(ASCII_Letter (ASCII_AlphaNumeric / "/" ASCII_AlphaNumeric)*)
```

Reply:
I shuffled it around a bit in b925d43482a05f61668e815a7653e86dec104b86. This also explicitly parses zero or one `/`s.